### ISSUE TITLE
PSR12.Classes.ClassInstantiation false positive when instantiating class from array index

### ISSUE THREAD

**Issue by TomasVotruba** [squizlabs/PHP_CodeSniffer#2047]
Version: 3.3.0

Sniff: PHP_CodeSniffer\Standards\PSR12\Sniffs\Classes\ClassInstantiationSniff.MissingParenthesis

```diff
             $configuration['layout'],
             $configuration['route_prefix'],
             $configuration['object'] ?? File::class,
-            isset($configuration['object_sorter']) ? new $configuration['object_sorter']() : new FileNameObjectSorter()
+            isset($configuration['object_sorter']) ? new $configuration()['object_sorter']() : new FileNameObjectSorter()

```

**Comment by gsherwood** [comment]
Thanks for reporting this. The sniff now supports array indexes properly.

**Comment by TomasVotruba** [comment]
Thanks for fast fix :1st_place_medal:
